### ISSUE TITLE
Support `numpy>=2`

### ISSUE THREAD

**Issue by paddyroddy** [astro-informatics/sleplet#389]
### Describe the Bug

Currently blocked by https://github.com/astro-informatics/ssht/issues/74

### To Reproduce

```python
import sleplet
...
```

### Expected Behaviour

-

### Actual Behaviour

-

### Version In Use

v1.4.10

### Additional Context

```markdown
- Python version: 3.11.9
- Operating system: macOS
```

**Comment by paddyroddy** [comment]
Closing in favour of #390
